src/overlays/actors/ovl_En_Guest/z_en_guest.c

```diff
@@ -20,23 +34,200 @@ const ActorInit En_Guest_InitVars = {
+        }
+    } else {
+        if (this->unk_2CA != 0) {
+            this->unk_2CA -= 1;
+        } else {
+            this->unk_30E += 1;
+            if (this->unk_30E >= 3) {
+                this->unk_30E = 0;
+                this->unk_30D = 0;
+                this->unk_2CA = (s32)Math_Rand_ZeroFloat(60.0f) + 20;
+            } else {
+                this->unk_2CA = 1;
+            }
+        }
+    }
+}
+
+void func_80A50518(EnGuest* this, GlobalContext* globalCtx) {
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        this->actionFunc = &func_80A5057C;
+    } else if (this->actor.xzDistanceFromLink < 100.0f) {
+        func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    }
+}
+
+void func_80A5057C(EnGuest* this, GlobalContext* globalCtx) {
+    if (func_8010BDBC(&globalCtx->msgCtx) == 6) {
+        if (func_80106BC8(globalCtx) != 0) {
+            this->actionFunc = &func_80A50518;
+        }
+    }
+}
+
+void func_80A505CC(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuest* this = THIS;
+    u32 padding;
+    Player* player;
+
+    player = PLAYER;
+    this->unk_2C8 += 1;
+
+    func_80A5046C(this);
+    this->actionFunc(this, globalCtx);
+
+    this->unk_2A0.unk_18 = player->actor.posRot.pos;
+    if (gSaveContext.linkAge == 0) {
+        this->unk_2A0.unk_14 = 10.0f;
+    } else {
+        this->unk_2A0.unk_14 = 20.0f;
+    }
+    func_80034A14(&this->actor, &this->unk_2A0, 6, 2);
+
+    func_80034F54(globalCtx, this->unk_2CC, this->unk_2EC, 16);
+
+    gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[this->objBankIndex].segment);
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Actor_SetHeight(&this->actor, 60.0f);
+
+    Collider_CylinderUpdate(&this->actor, &this->colliderCylinder);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colliderCylinder.base);
+}
+
+Gfx* func_80A50708(GraphicsContext* globalCtx, u8 r, u8 g, u8 b, u8 a) {
+    Gfx* temp_ret;
+
+    temp_ret = Graph_Alloc(globalCtx, 0x10);
+    gDPSetEnvColor(temp_ret, r, g, b, a);
+    gSPEndDisplayList(temp_ret + 1);
+
+    return temp_ret;
+}
+
+s32 func_80A50774(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnGuest* this = THIS;
+    Vec3s sp3C;
+    Gfx* dispRefs[5];
+
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_en_guest.c", 352);
+
+    if (limbIndex == 15) {
+        *dList = &D_060059B0;
+        Matrix_Translate(1400.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+        sp3C = this->unk_2A0.unk_08;
+        Matrix_RotateX((sp3C.y / 32768.0f) * M_PI, MTXMODE_APPLY);
+        Matrix_RotateZ((sp3C.x / 32768.0f) * M_PI, MTXMODE_APPLY);
+        Matrix_Translate(-1400.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Guest/EnGuest_Update.s")
+    if (limbIndex == 8) {
+        sp3C = this->unk_2A0.unk_0E;
+        Matrix_RotateX(((0 - sp3C.y) / 32768.0f) * M_PI, MTXMODE_APPLY);
```

Comment:
```suggestion
        Matrix_RotateX((-sp3C.y / 32768.0f) * M_PI, MTXMODE_APPLY);
```
only if this matches obviously